proof/crefine/ARM/Arch_C.thy

```diff
@@ -1885,6 +1883,129 @@ lemma performPageInvocationMapPDE_ccorres:
   apply (auto simp: unat_arith_simps unat_word_ariths)
   done
 
+lemma setMRs_single:
+  "setMRs thread buffer [val] = do _ \<leftarrow> asUser thread (setRegister register.R2 val);
+       return 1
+    od"
```

Comment:
was this copied from elsewhere? indentation is incorrect; if you want `do` on the rhs, the `_ <- asUser` should be on next line, the `od` aligned with `setMRs` and the `return` two from `od`.

Reply:
Also, interestingly, why does only ARM have this?